src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menArray = people.filter(object => object.sex === 'm');
+
+  const men = century
+    ? menArray.filter(object => Math.ceil(object.died / 100) === century)
+    : menArray;
+
+  const totalAge = men.reduce((sum, item) => {
+    const age = item.died - item.born;
+
+    return sum + age;
+  }, 0);
+
+  return totalAge / men.length;
```

Comment:
you should create function for reduce method, the same logic with reduce you have here and in `calculateWomenAverageAge` function